standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -270,8 +270,12 @@ public void alterTable(RawStore msdb, Warehouse wh, String catName, String dbnam
 
             // get new location
             assert(isReplicated == HMSHandler.isDbReplicationTarget(db));
-            if (renamedTranslatedToExternalTable) {
+            if (renamedTranslatedToExternalTable && !tableInSpecifiedLoc) {
               destPath = new Path(newt.getSd().getLocation());
```

Comment:
What happen if the path for the old table `b` is `file:${system:test.tmp.dir}/tmp.db/b`?  Does it throw the same exception?

Reply:
nit: we can build the `destPath` under `if (renamedTranslatedToExternalTablc)`:
```java
if (renamedTranslatedToExternalTablc) {
   if (tableInSpecifiedLoc) {
     ....
   } else {
    ...
   }
}

Reply:
@dengzhhu653 - if you explicitly specify a table location for old table b: as `file:${system:test.tmp.dir}/tmp.db/b` then it fails with the same exception. Only if you don't specify a location the query would pass